src/pal/inc/pal_mstypes.h

```diff
@@ -584,37 +584,29 @@ typedef LONG_PTR LPARAM;
 #define _PTRDIFF_T
 #endif
 
-#ifdef PAL_STDCPP_COMPAT
-
-#ifdef BIT64
-typedef unsigned long int uintptr_t;
-#else // !BIT64
-typedef unsigned int uintptr_t;
-#endif // !BIT64
-
-typedef char16_t WCHAR;
-
-#else // PAL_STDCPP_COMPAT
-
-typedef wchar_t WCHAR;
-#if defined(__linux__) 
+#ifndef PAL_STDCPP_COMPAT
+#if defined(__APPLE_CC__) || defined(__linux__)
 #ifdef BIT64
 typedef long int intptr_t;
 typedef unsigned long int uintptr_t;
 #else // !BIT64
 typedef int intptr_t;
 typedef unsigned int uintptr_t;
-#endif // !BIT64
+#endif // BIT64
 #else
 typedef INT_PTR intptr_t;
 typedef UINT_PTR uintptr_t;
 #endif
-
-#endif // PAL_STDCPP_COMPAT
-
+#endif // !PAL_STDCPP_COMPAT
 #define _INTPTR_T_DEFINED
 #define _UINTPTR_T_DEFINED
 
+#ifdef PAL_STDCPP_COMPAT
```

Comment:
@janvorli BTW, why WCHAR is wchar_t on Linux when PAL_STDCPP_COMPAT is not defiend? I thought you care about its size to be 16 bits.

Reply:
On Unix , when PAL_STDCPP_COMPAT is not defined, the wchar_t is also redefined, see src/pal/inc/pal_char16.h